web/src/settings_banner.ts

```diff
@@ -0,0 +1,170 @@
+import $ from "jquery";
+
+import * as banners from "./banners.ts";
+import type {Banner} from "./banners.ts";
+import {$t} from "./i18n.ts";
+import * as settings_data from "./settings_data.ts";
+
+const UPGRADE_ACCESS_BANNER: Banner = {
+    intent: "info",
+    label: $t({defaultMessage: "Available on Zulip Cloud Standard."}),
+    buttons: [
+        {
+            label: $t({defaultMessage: "Upgrade to access"}),
+            custom_classes: "request-upgrade",
+            attention: "quiet",
+        },
+    ],
+    close_button: false,
+};
+
+const UPGRADE_BANNER: Banner = {
+    intent: "neutral",
+    label: $t({defaultMessage: "Available on Zulip Cloud Standard."}),
+    buttons: [],
+    close_button: false,
+};
+
+const UPGRADE_OR_SPONSORSHIP_BANNER: Banner = {
+    intent: "info",
+    label: $t({
+        defaultMessage: "Available on Zulip Cloud Standard.",
+    }),
+    buttons: [
+        {
+            label: $t({defaultMessage: "Upgrade"}),
+            custom_classes: "request-upgrade",
+            attention: "quiet",
+        },
+        {
+            label: $t({defaultMessage: "Request sponsorship"}),
+            custom_classes: "request-sponsorship",
+            attention: "borderless",
+        },
+    ],
+    close_button: false,
+};
+
+const SPONSORSHIP_BANNER: Banner = {
+    intent: "neutral",
+    label: $t({defaultMessage: "Available on Zulip Cloud Standard."}),
+    buttons: [],
+    close_button: false,
```

Comment:
This and `UPGRADE_BANNER` are exactly same, maybe we can use the same object by having a better name.